src/LRUCache.jl

```diff
@@ -183,57 +181,54 @@ function Base.resize!(lru::LRU{K, V}; maxsize::Integer = lru.maxsize) where {K,
     lock(lru.lock) do
         _unsafe_resize!(lru, evictions, maxsize)
     end
-    _notify_evictions!(lru.eviction_callback, evictions)
+    _finalize_evicitions!(lru.finalizer, evictions)
     return lru
 end
 
 function Base.delete!(lru::LRU{K, V}, key) where {K, V}
-    evictions = Tuple{K, V}[]
-    lock(lru.lock) do
+    v = lock(lru.lock) do
         v, n, s = pop!(lru.dict, key)
-        if lru.eviction_callback !== _no_callback
-            push!(evictions, (key, v))
-        end
         lru.currentsize -= s
         _delete!(lru.keyset, n)
+        return v
+    end
+    if lru.finalizer !== nothing
+        lru.finalizer(key, v)
     end
-    _notify_evictions!(lru.eviction_callback, evictions)
     return lru
 end
 function Base.pop!(lru::LRU{K, V}, key) where {K, V}
-    evictions = Tuple{K, V}[]
-    v = lock(lru.lock) do
+    (key, v) = lock(lru.lock) do
         v, n, s = pop!(lru.dict, key)
-        if lru.eviction_callback !== _no_callback
-            push!(evictions, (key, v))
-        end
         lru.currentsize -= s
         _delete!(lru.keyset, n)
-        return v
+        return (key, v)
+    end
+    if lru.finalizer !== nothing
+        lru.finalizer(key, v)
     end
-    _notify_evictions!(lru.eviction_callback, evictions)
     return v
 end
 
 function Base.empty!(lru::LRU{K, V}) where {K, V}
     evictions = Tuple{K, V}[]
     lock(lru.lock) do
-        if lru.eviction_callback === _no_callback
+        if lru.finalizer === nothing
             lru.currentsize = 0
             empty!(lru.dict)
             empty!(lru.keyset)
         else
+            sizehint!(evictions, length(lru))
             _unsafe_resize!(lru, evictions, 0)
         end
     end
-    _notify_evictions!(lru.eviction_callback, evictions)
+    _finalize_evicitions!(lru.finalizer, evictions)
     return lru
 end
 
-function _notify_evictions!(callback, evictions)
-    while !isempty(evictions)
-        key, value = pop!(evictions)
-        callback(key, value)
+function _finalize_evicitions!(finalizer, evictions)
```

Comment:
Evictions is misspelled in this function name.

Also the definition of `_no_callback` can be removed, but github won't let me comment on lines outside the immediate context of the diff.

Reply:
Thanks for carefully checking this. That's what you get from doing "find and replace all" to quickly :-).